server/rest/src/main/java/org/apache/karaf/cave/server/rest/Service.java

```diff
@@ -47,9 +48,9 @@ public Service(CaveRepositoryService service) {
      * @throws Exception in case of creation failure.
      */
     @POST
-    @Consumes("application/xml")
-    @Produces("application/xml")
-    public Repository create(String name, boolean scan) throws Exception {
+    @Path("/repositories")
+    @Produces("application/json")
```

Comment:
We annotation with JAXB but we provide json output. Thoughts ?

Reply:
I use annotation with JAXB to define and force attribute name in the json output like "cave-repositories" :
```
@XmlRootElement(name = "cave-repository")
@XmlAccessorType(XmlAccessType.NONE)
@XmlType(name = "Repository", propOrder = {"name", "increment", "location", "resources"})
public class Repository {
...
```
It's not mandatory but I wouldn't change existing object name return but we can change it.

Reply:
If I remove JABX annotations, I've got this error :

No message body writer has been found for class [Lorg.apache.karaf.cave.server.rest.Repository;, ContentType: application/json

Reply:
We need to add a JSONProvider to the Servlet implementing CFXNonSpringServlet :
```
        String alias = getString("cave.rest.alias", "/cave/rest");
        Hashtable<String, String> config = new Hashtable<>();
        if (getConfiguration() != null) {
            for (Enumeration<String> e = getConfiguration().keys(); e.hasMoreElements();) {
                String key = e.nextElement();
                String val = getConfiguration().get(key).toString();
                config.put(key, val);
            }
        }
        this.alias = alias;
        this.servlet = new CaveRestServlet(service);
        this.httpService.registerServlet(this.alias, this.servlet, config, null);
```
I've already done this with blueprint but not in a java way, I can try to do it.

Reply:
OK, so it uses JABX -> JSON converter. That's fine as soon as we support both. Thanks.